tests/parse_shebang_test.py

```diff
@@ -91,6 +91,18 @@ def test_normexe_does_not_exist_sep():
     assert excinfo.value.args == ('Executable `./i-dont-exist-lol` not found',)
 
 
+def test_normexe_not_executable(in_tmpdir):
+    exe_path = os.path.abspath(
+        write_executable('/usr/bin/env sh', filename='run.myext'),
+    )
+    os.chmod(exe_path, 444)
```

Comment:
simpler:

```python
in_tmpdir.join('exe').ensure()
with pytest.raises(OSError) as excinfo:
    parse_shebang.normexe('./exe')
assert excinfo.value.args == ('Executable `./exe` is not executable',)
```

Reply:
Hmm, I get `AttributeError: 'str' object has no attribute 'ensure'` when I try to run that test.

Reply:
ah right, this is before I switched to using the `tmpdir` fixture more -- so instead:

```python
def test_...(tmpdir):
    tmpdir.join('exe').ensure()
    with tmpdir.as_cwd(), pytest.raises(OSError) as excinfo:
        parse_shebang.normexe('./exe')
    assert excinfo.value.args == ('Executable `./exe` is not executable',)
```